src/backend/query_builder.rs

```diff
@@ -58,6 +58,36 @@ pub trait QueryBuilder: QuotedBuilder {
         sql: &mut SqlWriter,
         collector: &mut dyn FnMut(Value),
     ) {
+        if let Some(_) = &select.exists {
+            // Clone current query and clean all eventual select to perform a wildcard select
+            let mut sub_query = select.clone();
+            sub_query
+                .clear_selects()
+                .expr(Expr::expr(SimpleExpr::Custom("*".to_string())))
+                .exists = None;
+
+            // Wrap the subquery into an `EXISTS()` statement
+            let mut sub_sql = SqlWriter::new();
+            write!(sub_sql, " EXISTS (").unwrap();
+            self.prepare_select_statement(&sub_query, &mut sub_sql, collector);
+            write!(sub_sql, ")").unwrap();
+
+            // Return a query aliased to exists
+            let main_query =
+                Query::select()
+                    .expr_as(
+                        Expr::expr(
+                            SimpleExpr::Custom(sub_sql.result())
+                        ),
+                        Alias::new("exists")
+                    )
+                    .take();
+
+            self.prepare_select_statement(&main_query, sql, collector);
+
+            return ()
+        }
```

Comment:
This `clone` seems unnecessary and nested calling of `prepare_select_statement` method could be avoided.

We could incorporate the logic of "exists statement" into the body of `prepare_select_statement` method.

The main difference between a normal select and "exists statement" are
1. all select are ignored and removed with an "*"
1. normal select wrapped with "SELECT EXISTS ( ... ) AS xxx"

Reply:
Before pushing this, I was manually writing the `SELECT EXISTS()` statement without cloning the `SelectStatement`, just cleaning the selects and wrapping it result into `String` using `write!` before and after to add `SELECT EXISTS(` and `) AS exists` and then abort with return ().

With this simpler solution we are still forced to make a nested call to `prepare_select_statement` to generate the subquery with all the where clauses and resetting the `exists` value.

Without nesting and aborting, this could lead the end-user into SQL errors like `SELECT EXISTS(SELECT * FROM "characters") ORDER BY "characters"."font_size"`.

Reply:
> Without nesting and aborting, this could lead the end-user into SQL errors like `SELECT EXISTS(SELECT * FROM "characters") ORDER BY "characters"."font_size"`.

Would it be possible to write the SQL like `SELECT EXISTS(SELECT * FROM "characters" ORDER BY "characters"."font_size")`? (where the closing parenthesis moved to the very end of original select)

Or, we could even ignore the order by expression if it's an exists statement.

Reply:
The query you provided is valid (inefficient but valid ^^'), with my "nested" implementation, this is already possible because I clean only the columns. This would be the result of this:
```rust
Query::select()
    .exists()
    .from(Char::Table)
    .and_where(Expr::col(Char::Character).eq("A"))
    .order_by(Character::Id, Order::Desc)
```
> Or, we could even ignore the order by expression if it's an exists statement.

Yeah, we could ignore the order by expression if it's an exists statement, but we would also need to ignore all the further preparations too, including the further `prepare_select_statement`, the order by was an exemple, but it's related to all of them, the distinct, the joins, offset, limits and so on. This is why I ended up just in just cleanly aborting.

As an exemple, I removed the aborting and rerun the test without modification, this is the result:
```sql
SELECT EXISTS (SELECT * FROM `character` WHERE `character` = 'A') AS `exists`SELECT FROM `character` WHERE `character` = 'A'
                                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The end section was appended because the current `SelectStatement` still includes table and conditions, we can get rid of the `SELECT` here by moving the block after `write!(sql, "SELECT ").unwrap();` and replacing our `SELECT EXISTS` by just `EXISTS` but we can't get rid of  the rest of the expression with:
- ignoring all blocks after
- or aborting

Reply:
I was thinking to refactor it into https://github.com/billy1624/sea-query/commit/aaf140e2f1e9395b5a68a9b2aa4b02801bbc579b to avoid `clone` and early exit

Reply:
Hmm, this seems cleaner indeed 😄

Reply:
You can cherrypick the useful bits loll